bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -1850,6 +1853,41 @@ private void setBackground(Color color, long handle) {
 	device.internal_dispose_GC(hDC, null);
 }
 
+private ImageData scaleImageData(Device device, int currentZoom, int targetZoom, ImageData imageData) {
```

Comment:
1. Should be `static`
2. Why change the order of the parameters? It only adds more lines to review and also makes the new method a non-drop-in replacement.

```suggestion
private static ImageData scaleImageData(Device device, final ImageData imageData, int targetZoom, int currentZoom) {
```

Reply:
Done

Reply:
But cannot make it static since we need some of the attributes of the image later in a method we call from this method.